### ISSUE TITLE
Unable to create an admin account after creating a new app with create-medusa-app

### ISSUE THREAD

**Issue by golaughlin** [medusajs/medusa#6400]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/medusajs/medusa/issues

### Issue Summary

When I create a new app using create-medusa-app, I am unable to get past the "Create Account Admin" page. When I press the "Create Account" button, It gives me a `can't access property 0, n is undefined` error.
<img width="1584" alt="Screenshot 2024-02-14 at 10 11 46 AM" src="https://github.com/medusajs/medusa/assets/17213463/a4268934-f5d0-433c-9acc-916c4466b7e4">

### How can this issue be resolved?

1.
2.
3.
...

### Are you interested in working on this issue?

- [ ] I would like to fix this issue

**Comment by shahednasser** [comment]
Can you check the console in the browser for any errors and share them?

**Comment by golaughlin** [comment]
Here are the screenshots of the browser console from Firefox Developer Edition 123.0b9 (64-bit) on macOS 14.2.1

<img width="1624" alt="Screenshot 2024-02-14 at 1 03 26 PM" src="https://github.com/medusajs/medusa/assets/17213463/03ffcab7-481f-4c8c-9da4-c5ba68975c40">
![Uploading Screenshot 2024-02-14 at 1.03.32 PM.p
<img width="1624" alt="Screenshot 2024-02-14 at 1 03 36 PM" src="https://github.com/medusajs/medusa/assets/17213463/a7126813-12cd-48f7-89dc-c3358826095f">
ng…]()
<img width="1624" alt="Screenshot 2024-02-14 at 1 03 42 PM" src="https://github.com/medusajs/medusa/assets/17213463/016da96f-541d-439a-aa16-cf42f9672b58">
<img width="1624" alt="Screenshot 2024-02-14 at 1 03 46 PM" src="https://github.com/medusajs/medusa/assets/17213463/b6fdc75b-69a3-4ded-bfb4-085cc7b1ac32">
<img width="1624" alt="Screenshot 2024-02-14 at 1 03 50 PM" src="https://github.com/medusajs/medusa/assets/17213463/d02d38ba-e60f-428e-86c8-a4263b11bbe8">

**Comment by golaughlin** [comment]
And here's what the NPM terminal showed when I did `create-medusa-app`
<img width="1624" alt="Screenshot 2024-02-14 at 1 07 32 PM" src="https://github.com/medusajs/medusa/assets/17213463/8e8946c2-6ee5-420f-a7bd-a0d3b3c7b9f8">

**Comment by shahednasser** [comment]
Thank you for sharing. Are you running this command on a remote server? Or do you have the `PORT` environment variable set to something other than `9000`? Seems like there's a problem reaching your Medusa backend from the admin. Can you try opening in your browser the URL `localhost:9000/store/products`?

**Comment by golaughlin** [comment]
So the `PORT` environment variable defaults to `7001` instead of 9000 when I run `medusa develop`. I have not changed the `PORT` number in the code. Trying to open the `http://localhost:9000/store/products` in the URL gives me a `
The connection was reset` error. Trying to open the `http://localhost:7001/store/products` in the URL gives me a 404 Not found error.

**Comment by shahednasser** [comment]
`7001` is the admin's port. The `PORT` environment variable is the backend's port which should be `9000`.

It seems like the `9000` port isn't exposed on your server so you're not able to access it. You should be able to access `localhost:9000/store/products` through the browser at the very least.

**Comment by zayaserg** [comment]
intresting, i created new medusa project from cli, and have this issue too
admin port 7001 working, but port 9000 not working on windows 11

**Comment by zayaserg** [comment]
@golaughlin you need to install postgresql and redis and uncomment lines with redis in medusa-config.js

**Comment by golaughlin** [comment]
So I went ahead and installed Redis and uncomment the lines with redis. I then created a new admin user with `medusa user --email <email> --password <password>` and restarted the medusa server and client with `medusa develop`. But when I try to login it gives me a `These credentials do not match our records.` error.

Here's what the `medusa develop` terminal told me...
```
yeen125@MacBook-Pro-2 my-medusa-store % medusa user --email golaughlin@medusa.test --password medusa
medusa user --email golaughlin@medusa.test --password medusa
[medusa-config] ⚠️ jwt_secret not found. fallback to either cookie_secret or default 'supersecret'.
[medusa-config] ⚠️ cookie_secret not found. fallback to either cookie_secret or default 'supersecret'.
info:    Connection to Redis established
✔ Models initialized – 12ms
✔ Plugin models initialized – 13ms
✔ Strategies initialized – 14ms
✔ Database initialized – 64ms
✔ Repositories initialized – 22ms
✔ Services initialized – 51ms
⠋ Initializing modules
✔ Modules initialized – 93ms
✔ Express intialized – 2ms
✔ Plugins intialized – 239ms
✔ Subscribers initialized – 5ms
✔ API initialized – 25ms
⠋ Initializing defaults
warn:    You don't have any notification provider plugins installed. You may want to add one to your project.
✔ Defaults initialized – 31ms
⠋ Initializing search engine indexing
✔ Indexing event emitted – 2ms
yeen125@MacBook-Pro-2 my-medusa-store % medusa develop
Successfully compiled 0 files with Babel (3ms).
[medusa-config] ⚠️ jwt_secret not found. fallback to either cookie_secret or default 'supersecret'.
[medusa-config] ⚠️ cookie_secret not found. fallback to either cookie_secret or default 'supersecret'.
info:    Connection to Redis established
✔ Models initialized – 10ms
✔ Plugin models initialized – 18ms
✔ Strategies initialized – 9ms
⠋ Initializing database
✔ Database initialized – 66ms
✔ Repositories initialized – 12ms
✔ Services initialized – 26ms
⠋ Initializing modules
✔ Modules initialized – 82ms
⠋ Initializing express
✔ Express intialized – 1ms
<i> [webpack-dev-server] Loopback: http://localhost:7001/
<i> [webpack-dev-server] On Your Network (IPv4): http://192.168.4.40:7001/
<i> [webpack-dev-server] On Your Network (IPv6): http://[fe80::1]:7001/
<i> [webpack-dev-server] Content not from webpack is served from '/Users/yeen125/Medusa Projects/my-medusa-store/public' directory
<i> [webpack-dev-server] 404s will fallback to '/'
⠋ Initializing plugins
✔ Plugins intialized – 159ms
✔ Subscribers initialized – 4ms
✔ API initialized – 14ms
⠋ Initializing defaults
warn:    You don't have any notification provider plugins installed. You may want to add one to your project.
✔ Defaults initialized – 39ms
⠋ Initializing search engine indexing
✔ Indexing event emitted – 8ms
✔ Server is ready on port: 9000 – 7ms
<i> [webpack-dev-middleware] wait until bundle finished: /
info:    Processing user.created which has 0 subscribers
assets by path *.js 53.7 MiB 25 assets
assets by path public/ 2.75 MiB
  asset public/locales/tm/translation.json 193 KiB [emitted] [from: node_modules/@medusajs/admin-ui/ui/public/locales/tm/translation.json] [copied]
  + 21 assets
assets by path *.ttf 1.24 MiB
  asset 442f6a4cc2f03aaf41f6.ttf 398 KiB [emitted] [immutable] [from: .cache/admin/src/fonts/Inter-SemiBold.ttf] (auxiliary name: main)
  + 4 assets
assets by path *.css 306 KiB
  asset main.css 230 KiB [emitted] (name: main)
  + 2 assets
assets by path images/*.svg 2.33 KiB
  asset images/icon-plus-gray.svg 1.46 KiB [emitted] [from: node_modules/react-nestable/dist/styles/icon-plus-gray.svg] (auxiliary id hint: vendors)
  asset images/icon-minus-gray.svg 884 bytes [emitted] [from: node_modules/react-nestable/dist/styles/icon-minus-gray.svg] (auxiliary id hint: vendors)
asset index.html 437 bytes [emitted]
Entrypoint main 10.8 MiB (1.24 MiB) = runtime~main.bundle.js 141 KiB main.css 230 KiB main.bundle.js 10.4 MiB 5 auxiliary assets
orphan modules 1.37 MiB (javascript) 1.24 MiB (asset) 15.2 KiB (runtime) [orphan] 659 modules
runtime modules 83.9 KiB 49 modules
modules by path ./node_modules/ 9.94 MiB (javascript) 26.8 KiB (css/mini-extract)
  javascript modules 9.94 MiB 2049 modules
  css modules 26.8 KiB
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./node_modules/react-nestable/dist/styles/index.css 1.76 KiB [built] [code generated]
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./node_modules/react-datepicker/dist/react-datepicker.css 25.1 KiB [built] [code generated]
  ./node_modules/axios/package.json 1.92 KiB [built] [code generated]
modules by path ./.cache/admin/src/ 8.7 MiB (javascript) 156 KiB (css/mini-extract)
  javascript modules 8.7 MiB 788 modules
  css modules 156 KiB
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./.cache/admin/src/assets/styles/global.css 156 KiB [built] [code generated]
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./.cache/admin/src/domain/product-categories/styles/product-categories.css 365 bytes [built] [code generated]
./util.inspect (ignored) 15 bytes [built] [code generated]
webpack 5.90.1 compiled successfully in 8221 ms
<i> [webpack-dev-middleware] wait until bundle finished: /main.bundle.js
assets by status 4.29 MiB [cached] 33 assets
assets by status 53.7 MiB [emitted]
  assets by chunk 18.6 MiB (id hint: vendors)
    asset vendors-node_modules_iso8601-duration_lib_index_js-node_modules_lodash_isEqual_js-node_module-b01569.chunk.js 8.5 MiB [emitted] [minimized] (id hint: vendors)
    asset vendors-node_modules_medusajs_ui_dist_cjs_index_js-node_modules_radix-ui_react-accordion_dist-1308db.chunk.js 8.03 MiB [emitted] [minimized] (id hint: vendors)
    asset vendors-node_modules_emoji-picker-react_dist_emoji-picker-react_esm_js-node_modules_react-col-bb12f4.chunk.js 1.5 MiB [emitted] [minimized] (id hint: vendors)
    asset tsub-middleware.chunk.js 400 KiB [emitted] [minimized] (name: tsub-middleware) (id hint: vendors)
    asset ajs-destination.chunk.js 79.6 KiB [emitted] [minimized] (name: ajs-destination) (id hint: vendors)
    asset vendors-node_modules_segment_analytics_js-video-plugins_dist_index_umd_js.chunk.js 53.5 KiB [emitted] [minimized] (id hint: vendors)
    asset vendors-node_modules_radix-ui_react-collapsible_dist_index_mjs-node_modules_radix-ui_react-co-1fa14b.chunk.js 45.5 KiB [emitted] [minimized] (id hint: vendors)
  assets by info 3.24 KiB [immutable]
    asset main.18cc647fe8cb2f221f3b.hot-update.js 2.19 KiB [emitted] [immutable] [hmr] [minimized] (name: main)
    asset runtime~main.18cc647fe8cb2f221f3b.hot-update.js 1.01 KiB [emitted] [immutable] [hmr] [minimized] (name: runtime~main)
    asset runtime_main.18cc647fe8cb2f221f3b.hot-update.json 43 bytes [emitted] [immutable] [hmr]
  + 18 assets
Entrypoint main 10.8 MiB (1.24 MiB) = runtime~main.bundle.js 141 KiB runtime~main.18cc647fe8cb2f221f3b.hot-update.js 1.01 KiB main.css 230 KiB main.bundle.js 10.4 MiB main.18cc647fe8cb2f221f3b.hot-update.js 2.19 KiB 5 auxiliary assets
cached modules 19.7 MiB (javascript) 183 KiB (css/mini-extract) 1.24 MiB (asset) 15.2 KiB (runtime) [cached] 3495 modules
runtime modules 38.4 KiB 21 modules
orphan modules 215 KiB [orphan] 1 module
./.cache/admin/src/assets/styles/global.css 335 bytes [built] [code generated]
webpack 5.90.1 compiled successfully in 2629 ms
```

And here's the screenshots of the login screen and the JavaScript console of the login screen:

<img width="1584" alt="Screenshot 2024-02-17 at 11 26 19 PM" src="https://github.com/medusajs/medusa/assets/17213463/7c2074b3-ba46-4ad7-8546-a138d1f23f19">

<img width="796" alt="Screenshot 2024-02-17 at 11 27 40 PM" src="https://github.com/medusajs/medusa/assets/17213463/341774d6-dff1-4af6-b186-618ee48d3347">

**Comment by zayaserg-reis** [comment]
@golaughlin  probably your user wasn't created 
did you run the migrations after project install?
npx @medusajs/medusa migrations run
npx @medusajs/medusa user -e 'email@example.com' -p 'some-password'

**Comment by shahednasser** [comment]
Just want to point out the installing Redis doesn't change anything, redis is only needed for features like Scheduled Jobs. So, your problem isn't really fixed because of that.

You can try running the new user command `npx medusa user -e admin@medusajs.com -p supersecret` and try logging in with that. If you still get the same error, then you still have the same problems with the ports.

**Comment by golaughlin** [comment]
So I have tried running migrations and creating a new user with `medusa user -e admin@medusajs.com -p medusa` but I still get the same `These credentials do not match our records` error.

I do not see anything on the console that suggests a port error.
```
yeen125@MacBook-Pro-2 my-medusa-store % medusa develop
Successfully compiled 0 files with Babel (3ms).
[medusa-config] ⚠️ jwt_secret not found. fallback to either cookie_secret or default 'supersecret'.
[medusa-config] ⚠️ cookie_secret not found. fallback to either cookie_secret or default 'supersecret'.
info:    Connection to Redis established
✔ Models initialized – 12ms
✔ Plugin models initialized – 15ms
✔ Strategies initialized – 16ms
⠋ Initializing database
✔ Database initialized – 75ms
✔ Repositories initialized – 26ms
✔ Services initialized – 49ms
⠋ Initializing modules
Local Event Bus installed. This is not recommended for production.
<i> [webpack-dev-server] Project is running at:
<i> [webpack-dev-server] Loopback: http://localhost:7001/
<i> [webpack-dev-server] On Your Network (IPv4): http://192.168.4.40:7001/
<i> [webpack-dev-server] On Your Network (IPv6): http://[fe80::1]:7001/
<i> [webpack-dev-server] Content not from webpack is served from '/Users/yeen125/Medusa Projects/my-medusa-store/public' directory
✔ Modules initialized – 88ms
✔ Express intialized – 2ms
⠋ Initializing plugins
✔ Plugins intialized – 166ms
✔ Subscribers initialized – 6ms
✔ API initialized – 29ms
⠋ Initializing defaults
warn:    You don't have any notification provider plugins installed. You may want to add one to your project.
✔ Defaults initialized – 53ms
⠋ Initializing search engine indexing
✔ Indexing event emitted – 2ms
✔ Server is ready on port: 9000 – 4ms
<i> [webpack-dev-middleware] wait until bundle finished: /
info:    Processing user.created which has 0 subscribers
assets by path *.js 53.7 MiB 25 assets
assets by path public/ 2.75 MiB
  asset public/locales/tm/translation.json 193 KiB [emitted] [from: node_modules/@medusajs/admin-ui/ui/public/locales/tm/translation.json] [copied]
  + 21 assets
assets by path *.ttf 1.24 MiB
  asset 442f6a4cc2f03aaf41f6.ttf 398 KiB [emitted] [immutable] [from: .cache/admin/src/fonts/Inter-SemiBold.ttf] (auxiliary name: main)
  + 4 assets
assets by path *.css 306 KiB
  asset main.css 230 KiB [emitted] (name: main)
  + 2 assets
assets by path images/*.svg 2.33 KiB
  asset images/icon-plus-gray.svg 1.46 KiB [emitted] [from: node_modules/react-nestable/dist/styles/icon-plus-gray.svg] (auxiliary id hint: vendors)
  asset images/icon-minus-gray.svg 884 bytes [emitted] [from: node_modules/react-nestable/dist/styles/icon-minus-gray.svg] (auxiliary id hint: vendors)
asset index.html 437 bytes [emitted]
Entrypoint main 10.8 MiB (1.24 MiB) = runtime~main.bundle.js 141 KiB main.css 230 KiB main.bundle.js 10.4 MiB 5 auxiliary assets
orphan modules 1.37 MiB (javascript) 1.24 MiB (asset) 15.2 KiB (runtime) [orphan] 659 modules
runtime modules 83.9 KiB 49 modules
modules by path ./node_modules/ 9.94 MiB (javascript) 26.8 KiB (css/mini-extract)
  javascript modules 9.94 MiB 2049 modules
  css modules 26.8 KiB
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./node_modules/react-nestable/dist/styles/index.css 1.76 KiB [built] [code generated]
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./node_modules/react-datepicker/dist/react-datepicker.css 25.1 KiB [built] [code generated]
  ./node_modules/axios/package.json 1.92 KiB [built] [code generated]
modules by path ./.cache/admin/src/ 8.7 MiB (javascript) 156 KiB (css/mini-extract)
  javascript modules 8.7 MiB 788 modules
  css modules 156 KiB
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./.cache/admin/src/assets/styles/global.css 156 KiB [built] [code generated]
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./.cache/admin/src/domain/product-categories/styles/product-categories.css 365 bytes [built] [code generated]
./util.inspect (ignored) 15 bytes [built] [code generated]
webpack 5.90.1 compiled successfully in 8145 ms
<i> [webpack-dev-middleware] wait until bundle finished: /logo.svg
<i> [webpack-dev-middleware] wait until bundle finished: /vendors-node_modules_radix-ui_react-collapsible_dist_index_mjs-node_modules_radix-ui_react-co-1fa14b.chunk.js
<i> [webpack-dev-middleware] wait until bundle finished: /vendors-node_modules_medusajs_ui_dist_cjs_index_js-node_modules_radix-ui_react-accordion_dist-1308db.chunk.js
<i> [webpack-dev-middleware] wait until bundle finished: /_cache_admin_src_extensions__main-entry_ts.chunk.js
<i> [webpack-dev-middleware] wait until bundle finished: /public/locales/en/translation.json
assets by status 4.29 MiB [cached] 33 assets
assets by status 53.7 MiB [emitted]
  assets by chunk 18.6 MiB (id hint: vendors)
    asset vendors-node_modules_iso8601-duration_lib_index_js-node_modules_lodash_isEqual_js-node_module-b01569.chunk.js 8.5 MiB [emitted] [minimized] (id hint: vendors)
    asset vendors-node_modules_medusajs_ui_dist_cjs_index_js-node_modules_radix-ui_react-accordion_dist-1308db.chunk.js 8.03 MiB [emitted] [minimized] (id hint: vendors)
    asset vendors-node_modules_emoji-picker-react_dist_emoji-picker-react_esm_js-node_modules_react-col-bb12f4.chunk.js 1.5 MiB [emitted] [minimized] (id hint: vendors)
    asset tsub-middleware.chunk.js 400 KiB [emitted] [minimized] (name: tsub-middleware) (id hint: vendors)
    asset ajs-destination.chunk.js 79.6 KiB [emitted] [minimized] (name: ajs-destination) (id hint: vendors)
    asset vendors-node_modules_segment_analytics_js-video-plugins_dist_index_umd_js.chunk.js 53.5 KiB [emitted] [minimized] (id hint: vendors)
    asset vendors-node_modules_radix-ui_react-collapsible_dist_index_mjs-node_modules_radix-ui_react-co-1fa14b.chunk.js 45.5 KiB [emitted] [minimized] (id hint: vendors)
  assets by info 3.24 KiB [immutable]
    asset main.e7f59d8317e8ed7e499c.hot-update.js 2.19 KiB [emitted] [immutable] [hmr] [minimized] (name: main)
    asset runtime~main.e7f59d8317e8ed7e499c.hot-update.js 1.01 KiB [emitted] [immutable] [hmr] [minimized] (name: runtime~main)
    asset runtime_main.e7f59d8317e8ed7e499c.hot-update.json 43 bytes [emitted] [immutable] [hmr]
  + 18 assets
Entrypoint main 10.8 MiB (1.24 MiB) = runtime~main.bundle.js 141 KiB runtime~main.e7f59d8317e8ed7e499c.hot-update.js 1.01 KiB main.css 230 KiB main.bundle.js 10.4 MiB main.e7f59d8317e8ed7e499c.hot-update.js 2.19 KiB 5 auxiliary assets
cached modules 19.7 MiB (javascript) 183 KiB (css/mini-extract) 1.24 MiB (asset) 15.2 KiB (runtime) [cached] 3495 modules
runtime modules 38.4 KiB 21 modules
orphan modules 215 KiB [orphan] 1 module
./.cache/admin/src/assets/styles/global.css 335 bytes [built] [code generated]
webpack 5.90.1 compiled successfully in 2695 ms
```

**Comment by shahednasser** [comment]
When you try to login, do you see a new request logged in the Medusa backend? If not, then the admin isn't able to send requests to the backend in the first place. Again, could be that you need to enable port forwarding for port `9000` or something similar in your server/system

**Comment by golaughlin** [comment]
I did not see a new request logged in. I did go ahead and retried the `npx create-medusa-app@latest` command but still got the same `cannot access property 0, n is undefined` error as before.
Here's what I got in my terminal.

```
yeen125@Helena Medusa Projects % sudo rm -r my-medusa-store
Password:
yeen125@Helena Medusa Projects % npx create-medusa-app@latest
? What's the name of your project? my-medusa-store
? Enter an email for your admin dashboard user admin@medusa-test.com
? Would you like to create the Next.js storefront? You can also create it later No
🚀 Starting project setup, this may take a few minutes.
✔ Created project directory
✔ Database medusa-g7Xo created
✔ Installed Dependencies
✔ Project Built
✔ Ran Migrations
✔ Created admin user
✔ Finished Preparation
✔ Project Prepared
Starting Medusa...
Successfully compiled 0 files with Babel (2ms).
6:22:47 PM [@medusajs/admin] Started development server on http://localhost:7001/
[medusa-config] ⚠️ redis_url not found. A fake redis instance will be used.
[medusa-config] ⚠️ jwt_secret not found. fallback to either cookie_secret or default 'supersecret'.
[medusa-config] ⚠️ cookie_secret not found. fallback to either cookie_secret or default 'supersecret'.
info:    Using fake Redis
6:22:48 PM [@medusajs/admin] Running in development mode. Skipping setup.
warn:    You don't have Redis configured. Scheduled jobs will not be enabled.
warn:    You don't have any notification provider plugins installed. You may want to add one to your project.
warn:    You don't have any taxProviders provider plugins installed. You may want to add one to your project.
warn:    No search engine provider was found: make sure to include a search plugin to enable searching
::1 - - [22/Feb/2024:00:22:48 +0000] "HEAD /health HTTP/1.1" 200 2 "-" "axios/1.6.7"
info:    Processing sales_channel.created which has 0 subscribers
info:    Processing invite.created which has 0 subscribers
info:    Processing region.created which has 0 subscribers
info:    Processing region.created which has 0 subscribers
info:    Processing publishable_api_key.created which has 0 subscribers
assets by path *.js 53.8 MiB 25 assets
assets by path public/ 2.75 MiB
  asset public/locales/tm/translation.json 193 KiB [emitted] [from: node_modules/@medusajs/admin-ui/ui/public/locales/tm/translation.json] [copied]
  + 21 assets
assets by path *.ttf 1.24 MiB
  asset 442f6a4cc2f03aaf41f6.ttf 398 KiB [emitted] [immutable] [from: .cache/admin/src/fonts/Inter-SemiBold.ttf] (auxiliary name: main)
  + 4 assets
assets by path *.css 306 KiB
  asset main.css 230 KiB [emitted] (name: main)
  + 2 assets
assets by path images/*.svg 2.33 KiB
  asset images/icon-plus-gray.svg 1.46 KiB [emitted] [from: node_modules/react-nestable/dist/styles/icon-plus-gray.svg] (auxiliary id hint: vendors)
  asset images/icon-minus-gray.svg 884 bytes [emitted] [from: node_modules/react-nestable/dist/styles/icon-minus-gray.svg] (auxiliary id hint: vendors)
asset index.html 437 bytes [emitted]
Entrypoint main 10.8 MiB (1.24 MiB) = runtime~main.bundle.js 141 KiB main.css 230 KiB main.bundle.js 10.4 MiB 5 auxiliary assets
orphan modules 1.37 MiB (javascript) 1.24 MiB (asset) 15.2 KiB (runtime) [orphan] 659 modules
runtime modules 84 KiB 49 modules
modules by path ./node_modules/ 9.95 MiB (javascript) 26.8 KiB (css/mini-extract)
  javascript modules 9.94 MiB 2117 modules
  css modules 26.8 KiB
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./node_modules/react-nestable/dist/styles/index.css 1.76 KiB [built] [code generated]
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./node_modules/react-datepicker/dist/react-datepicker.css 25.1 KiB [built] [code generated]
  ./node_modules/axios/package.json 1.92 KiB [built] [code generated]
modules by path ./.cache/admin/src/ 8.7 MiB (javascript) 156 KiB (css/mini-extract)
  javascript modules 8.7 MiB 788 modules
  css modules 156 KiB
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./.cache/admin/src/assets/styles/global.css 156 KiB [built] [code generated]
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./.cache/admin/src/domain/product-categories/styles/product-categories.css 365 bytes [built] [code generated]
./util.inspect (ignored) 15 bytes [built] [code generated]
webpack 5.90.3 compiled successfully in 7611 ms
assets by status 4.29 MiB [cached] 33 assets
assets by status 53.8 MiB [emitted]
  assets by chunk 18.7 MiB (id hint: vendors)
    asset vendors-node_modules_iso8601-duration_lib_index_js-node_modules_lodash_isEqual_js-node_module-b01569.chunk.js 8.5 MiB [emitted] [minimized] (id hint: vendors)
    asset vendors-node_modules_medusajs_ui_dist_cjs_index_js-node_modules_radix-ui_react-accordion_dist-1308db.chunk.js 8.09 MiB [emitted] [minimized] (id hint: vendors)
    asset vendors-node_modules_emoji-picker-react_dist_emoji-picker-react_esm_js-node_modules_react-col-bb12f4.chunk.js 1.5 MiB [emitted] [minimized] (id hint: vendors)
    asset tsub-middleware.chunk.js 400 KiB [emitted] [minimized] (name: tsub-middleware) (id hint: vendors)
    asset ajs-destination.chunk.js 79.6 KiB [emitted] [minimized] (name: ajs-destination) (id hint: vendors)
    asset vendors-node_modules_segment_analytics_js-video-plugins_dist_index_umd_js.chunk.js 53.5 KiB [emitted] [minimized] (id hint: vendors)
    asset vendors-node_modules_radix-ui_react-collapsible_dist_index_mjs-node_modules_radix-ui_react-co-1fa14b.chunk.js 45.5 KiB [emitted] [minimized] (id hint: vendors)
  assets by info 3.24 KiB [immutable]
    asset main.23a4e9a9ea8618c5ff04.hot-update.js 2.19 KiB [emitted] [immutable] [hmr] [minimized] (name: main)
    asset runtime~main.23a4e9a9ea8618c5ff04.hot-update.js 1.01 KiB [emitted] [immutable] [hmr] [minimized] (name: runtime~main)
    asset runtime_main.23a4e9a9ea8618c5ff04.hot-update.json 43 bytes [emitted] [immutable] [hmr]
  + 18 assets
Entrypoint main 10.8 MiB (1.24 MiB) = runtime~main.bundle.js 141 KiB runtime~main.23a4e9a9ea8618c5ff04.hot-update.js 1.01 KiB main.css 230 KiB main.bundle.js 10.4 MiB main.23a4e9a9ea8618c5ff04.hot-update.js 2.19 KiB 5 auxiliary assets
cached modules 19.7 MiB (javascript) 183 KiB (css/mini-extract) 1.24 MiB (asset) 15.2 KiB (runtime) [cached] 3563 modules
runtime modules 38.5 KiB 21 modules
orphan modules 215 KiB [orphan] 1 module
./.cache/admin/src/assets/styles/global.css 335 bytes [built] [code generated]
webpack 5.90.3 compiled successfully in 2618 ms
```

**Comment by golaughlin** [comment]
Here's the TypeError in more detail:

```
can't access property 0, n is undefined
i@http://localhost:7001/main.bundle.js:689:516
63281/A/r/<@http://localhost:7001/_cache_admin_src_pages_invite_tsx.chunk.js:5:5570
63281/A/r/o/</<@http://localhost:7001/_cache_admin_src_pages_invite_tsx.chunk.js:5:4741
63281/A/r/o/<@http://localhost:7001/_cache_admin_src_pages_invite_tsx.chunk.js:5:4846
D@http://localhost:7001/_cache_admin_src_pages_invite_tsx.chunk.js:5:1239
o@http://localhost:7001/_cache_admin_src_pages_invite_tsx.chunk.js:5:5749
```

This is after running the `npx create-medusa-app@latest` command without any other commands or changes to the code. I get the error whewn I try to create a password for the account.

**Comment by shahednasser** [comment]
If nothing is being logged in the terminal when you try to log-in, then the request isn't reaching the backend in the first place. Again this probably has to do with port forwarding or port not exposed by your server.

Can you try accessing `localhost:9000/store/products` in your browser?

**Comment by craftmyne** [comment]
Just wanna bump this, as I believe I am running into the same issue.

as I understand it if you set this up on a local machine it should open a window with an automatic invite link, what do you do if you set it up remotely?

Im setting up the server on a external machine but connecting via its local ip
I am able to access 192.168.0.2:9000/store/products, yielding the following:

![image](https://github.com/medusajs/medusa/assets/32849629/ec11351f-6698-4b56-b33a-e339b8e4b0e5)

but if I try to generate a invite link and use it i get the type error 

TypeError
n is undefined
Call Stack
 i
  main.bundle.js:689:516
 63281/k/t/<
  _cache_admin_src_pages_invite_tsx.chunk.js:5:5570
 63281/k/t/o/
  _cache_admin_src_pages_invite_tsx.chunk.js:5:4741
 63281/k/t/o/<
  _cache_admin_src_pages_invite_tsx.chunk.js:5:4846
 M
  _cache_admin_src_pages_invite_tsx.chunk.js:5:1239
 o
  _cache_admin_src_pages_invite_tsx.chunk.js:5:5749

**Comment by shahednasser** [comment]
If you check the [comment options](https://docs.medusajs.com/create-medusa-app#step-1-run-create-medusa-app) in the docs, you'll find a `--no-browser` option which shows you the link instead of opening the browser (however, you'd still need to run the Medusa backend manually)

> I am able to access 192.168.0.2:9000/store/products, yielding the following:

What's the admin URL? If it's also 192.168.0.2:7000 then you need to change the [admin_cors](https://docs.medusajs.com/development/backend/configurations#admin_cors) configuration to include it

**Comment by craftmyne** [comment]
I mean I'm able to access it without changing the admin cors to it, thats just its local ip on my network. Or should I change it to http://192.168.0.91:7000,http://192.168.0.91:7001

using the given invite link to create an account yields this

![image](https://github.com/medusajs/medusa/assets/32849629/6f58c47b-0ee3-40b4-ab6a-113da7bcf143)

![image](https://github.com/medusajs/medusa/assets/32849629/0f26d2ac-ebe1-4937-990e-57dd371cb46f)

if I access the ip to the products page I get  this output in the commandline, so I can access the backend, but it fails at creating an account

::ffff:192.168.0.246 - - [23/Feb/2024:12:12:15 +0000] "GET /store/products HTTP/1.1" 304 - "-" "Mozilla/5.0 (X11; Linux x86_64; rv:122.0) Gecko/20100101 Firefox/122.0"

**Comment by shahednasser** [comment]
Yes you need to change the CORS to `http://192.168.0.91:7001`. It's not about accessing the admin, it's about the admin being able to send requests to the backend.

**Comment by craftmyne** [comment]
Like so?

![image](https://github.com/medusajs/medusa/assets/32849629/27d8d5c7-363c-48e7-b96e-c6809c005bd0)

this yields the same error

I tried making a new user using npx medusa user -e some@email.com -p somepassword

![image](https://github.com/medusajs/medusa/assets/32849629/2983d2a1-27af-4534-852a-76f9878c2cf9)

which was seen by the backend but when I try to login it says the credentials don't match

![image](https://github.com/medusajs/medusa/assets/32849629/6159be48-5f6c-464a-b7ed-b13bbe75d150)


![image](https://github.com/medusajs/medusa/assets/32849629/9cd2d839-505d-49ca-87c9-8cdd23aa146d)

**Comment by golaughlin** [comment]
I am able to access `localhost:9000/store/products` using Postman and `curl localhost:9000/store/products`. However, Firefox gives a `The connection was reset` error.

**Comment by shahednasser** [comment]
@craftmyne looking at your log message:

> ::ffff:192.168.0.246 - - [23/Feb/2024:12:12:15 +0000] "GET /store/products HTTP/1.1" 304 - "-" "Mozilla/5.0 (X11; Linux x86_64; rv:122.0) Gecko/20100101 Firefox/122.0"

It shows the IP as `192.168.0.246`. Are you accessing from a different IP?

**Comment by craftmyne** [comment]
> It shows the IP as `192.168.0.246`. Are you accessing from a different IP?

Yeah I am accessing it from a different computer on the local network, are there any changes I should make if I do that?

**Comment by shahednasser** [comment]
Can you try adding your IP to the CORS as well? Also, can you show the backend logs when you try to login? Is the request reaching the backend?

**Comment by onu941** [comment]
@golaughlin @craftmyne i ran into this problem just now too and resolved it by changing my default browser from firefox to chrome.

**Comment by ShawnLiUVA** [comment]
Any solutions or suggestions? still got the same issue when following the installation guide..

**Comment by BernardMcWeeney** [comment]
I have the same issue:

Trying to install medusa js using npx create-medusa-app@latest on Hetzner cloud server ubuntu 22

I can access ipaddress:7001/invite?token=

but I get:

GET http://localhost:9000/admin/batch-jobs?limit=100 net::ERR_FAILED
or
GET http://localhost:9000/admin/auth net::ERR_FAILED
or
Access to XMLHttpRequest at 'http://localhost:9000/admin/batch-jobs?limit=100' from origin 'http://ipaddress:7001' has been blocked by CORS policy: The request client is not a secure context and the resource is in more-private address space `local`.

.env:
STORE_CORS=http://localhost:8000,http://localhost:7001
CACHE_REDIS_URL=http://localhost:6379

medusa-config:
// CORS when consuming Medusa from admin
const ADMIN_CORS =
  process.env.ADMIN_CORS || "http://localhost:7000,http://localhost:7001";

// CORS to avoid issues when consuming Medusa from a client
const STORE_CORS = process.env.STORE_CORS || "http://localhost:8000";

const DATABASE_URL =
  process.env.DATABASE_URL || "postgres://localhost/medusa-starter-default";

const REDIS_URL = process.env.REDIS_URL || "redis://localhost:6379";

**Comment by Ayush-v** [comment]
I am facing kind of similar issue i am not able to login to the db because there is no user created in the db 

the thing is i just found about this today and wanted to explore is but i am not able to run it properly and facing some issues
i tried seeding the database as well and it is stuck on loading "intializing express"

and other issue is i am not able to complete the setup properly 

```ssh
➜ Desktop npx create-medusa-app@latest
? What's the name of your project? my-medusa-store
? Enter an email for your admin dashboard user admin@medusa-test.com
? Would you like to create the Next.js storefront? You can also create it later No
🚀 Starting project setup, this may take a few minutes.
✔ Created project directory
✔ Database medusa-6vp8 created
✔ Installed Dependencies
✔ Project Built
✔ Ran Migrations
An error occurred while preparing project: Error: Command failed: npx @medusajs/medusa-cli@latest user -e admin@medusa-test.com --invite
- Initializing models

✔ Models initialized – 15ms
- Initializing plugin models

✔ Plugin models initialized – 11ms
- Initializing strategies

✔ Strategies initialized – 13ms
- Initializing database

✔ Database initialized – 71ms
- Initializing repositories

✔ Repositories initialized – 20ms
- Initializing services

✔ Services initialized – 50ms
- Initializing modules

✔ Modules initialized – 63ms
- Initializing express

Tue, 14 May 2024 08:13:40 GMT express-session deprecated req.secret; provide secret option at node_modules/@medusajs/medusa/dist/loaders/express.js:88:51
TypeError: JwtStrategy requires a secret or key
    at new JwtStrategy (/Users/ayush/Desktop/my-medusa-store/node_modules/passport-jwt/lib/strategy.js:45:15)
    at /Users/ayush/Desktop/my-medusa-store/node_modules/@medusajs/medusa/dist/loaders/passport.js:132:52
    at step (/Users/ayush/Desktop/my-medusa-store/node_modules/@medusajs/medusa/dist/loaders/passport.js:33:23)
    at Object.next (/Users/ayush/Desktop/my-medusa-store/node_modules/@medusajs/medusa/dist/loaders/passport.js:14:53)
    at /Users/ayush/Desktop/my-medusa-store/node_modules/@medusajs/medusa/dist/loaders/passport.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/ayush/Desktop/my-medusa-store/node_modules/@medusajs/medusa/dist/loaders/passport.js:4:12)
    at /Users/ayush/Desktop/my-medusa-store/node_modules/@medusajs/medusa/dist/loaders/passport.js:48:12
    at /Users/ayush/Desktop/my-medusa-store/node_modules/@medusajs/medusa/dist/loaders/index.js:427:65
    at step (/Users/ayush/Desktop/my-medusa-store/node_modules/@medusajs/medusa/dist/loaders/index.js:56:23)

```

**Comment by shahednasser** [comment]
Hi @Ayush-v this issue should be fixed now

**Comment by shahednasser** [comment]
Can you try again now? Issue should be fixed @Ayush-v

**Comment by Ayush-v** [comment]
Thank You!!
It is working now @shahednasser

**Comment by benedictgs** [comment]
facing similar issue. 

followed the process from [https://docs.medusajs.com/create-medusa-app](url)

tried on postgres local and dockerized postgres.

result in error accessing admin.

```
 sudo lsof -i :7001
[sudo] password for test:
COMMAND   PID USER   FD   TYPE  DEVICE SIZE/OFF NODE NAME
node    32236 test   31u  IPv4 1759631      0t0  TCP localhost:7001 (LISTEN)
```

able to `curl localhost:9000/store/products`
with appropriate response

```
 curl localhost:7001
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/logo.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Medusa</title>
  <script defer src="/runtime~main.bundle.js"></script><script defer src="/main.bundle.js"></script><link href="/main.css" rel="stylesheet"></head>
  <body>
    <div id="root"></div>
  </body>
</html>
```

testing other browser doesn't help contrary to mentioned.

> @golaughlin @craftmyne i ran into this problem just now too and resolved it by changing my default browser from firefox to chrome.

tried on firefox : connection was reset
on opera&chrome: ERR_EMPTY_RESPONSE

current config.
```
// CORS when consuming Medusa from admin
const ADMIN_CORS =
  process.env.ADMIN_CORS || "http://localhost:7000,http://localhost:7001";

// CORS to avoid issues when consuming Medusa from a client
const STORE_CORS = process.env.STORE_CORS || "http://localhost:8000";

const DATABASE_URL =
  process.env.DATABASE_URL || "postgres://localhost/medusa-starter-default";

const REDIS_URL = process.env.REDIS_URL || "redis://localhost:6379";
```




> If you check the [comment options](https://docs.medusajs.com/create-medusa-app#step-1-run-create-medusa-app) in the docs, you'll find a `--no-browser` option which shows you the link instead of opening the browser (however, you'd still need to run the Medusa backend manually)
> 
> > I am able to access 192.168.0.2:9000/store/products, yielding the following:
> 
> What's the admin URL? If it's also 192.168.0.2:7000 then you need to change the [admin_cors](https://docs.medusajs.com/development/backend/configurations#admin_cors) configuration to include it

` pnpm dlx create-medusa-app@latest --seed --no-browser`
tried deployment with above parameter end with same result.

**Comment by shahednasser** [comment]
> result in error accessing admin.

Can you clarify what the error is?

**Comment by sradevski** [comment]
Hey, thanks for the report! Since v2 brought a lot of architectural and API changes on the backend, we will be closing this ticket since it no longer applies to our new setup, or the issue has already been fixed. If you are still facing issues with v1,
please open a new ticket and we will address it as soon as possible. Thanks! 🙏

**Comment by haf** [comment]
The medusa CLI doesn't create admins, but the seed file does.

<img width="226" alt="image" src="https://github.com/user-attachments/assets/73155dcd-936e-4385-a5bd-10c89582b887">

**Comment by rickstonerz** [comment]
still is an issue.. luls.
how.... is this still an issue. been days on wondering why this issue exists.

**Comment by haf** [comment]
Yes this should probably be fixed for v1.x since it’s such important functionality

**Comment by p-pey** [comment]
> Just want to point out the installing Redis doesn't change anything, redis is only needed for features like Scheduled Jobs. So, your problem isn't really fixed because of that.
> 
> You can try running the new user command `npx medusa user -e admin@medusajs.com -p supersecret` and try logging in with that. If you still get the same error, then you still have the same problems with the ports.

try this command:

npx medusa user -e "youremail@gmail.com" -p "your-password"

**Comment by adamBoualleiguie** [comment]
@p-pey   same that npx command create user that is not admin
